pravega/pravega-operator
# Issue 388: Adding support for helm upgrades

Signed-off-by: SrishT <Srishti.Thakkar@dell.com>

### Change log description
To add support to upgrade pravega operator from version `0.4.x` to `0.5.0` via helm.

### Purpose of the change
Fixes #388 

### How to verify it
The user needs to follow the given steps in order to trigger the upgrade
- Install bookkeeper operator and cert-manager
- Install the issuer and certificate using the manifest `deploy/certificate.yaml` and plug the value `tls.crt` from the secret created by the certificate that has been created in this step using the following command
```
$ kubectl get secret selfsigned-cert-tls -o yaml | grep tls.crt
```
Here `selfsigned-cert-tls` is the name of the secret used in the file `deploy/certificate.yaml`
The `tls.crt` value so obtained needs to be provided either in the `webhookCert.crt` field of the values.yaml file of the 0.5.0 operator charts or in the `--set webhookCert.crt=<tls.crt>`  of the helm upgrade command.
- Execute the script `./scripts/pre-upgrade.sh <operator release name> <operator release namespace>`
- Trigger the operator upgrade. To trigger the upgrade, run the following command
```
$ helm upgrade <operator release name> <location of 0.5.0 charts> --set webhookCert.crt=<tls.crt>
```

You will see the following message as soon as the operator upgrade is done
```
Release <operator release name> has been upgraded. Happy Helming!
NAME: <operator release name>
LAST DEPLOYED: Thu Jun 25 14:21:02 2020
NAMESPACE: <operator release namespace>
STATUS: deployed
REVISION: 2
TEST SUITE: None
```

Soon after this, the conversion of the Pravega CR from from `v1alpha1` to `v1beta1` is triggered and the ownership of the Bookkeeper CR is migrated from the Pravega Operator to the Bookkeeper Operator.
The presence of the following log message in the Pravega Operator logs indicates the successful completion of the Pravega Operator version migration.
```
Version migration completed successfully
```
